src/bwamem.cpp

```
@@ -665,6 +710,28 @@ void mem_flt_chained_seeds(const mem_opt_t *opt, const bntseq_t *bns, const uint
         // workload. Off (min_ext_len==0): no-op, output byte-identical.
         mem_chain_drop_short_seeds(c, opt->min_ext_len);
 
+        /* D3 (--meth, PR-4): swap to the original read + per-hypothesis matrix. */
+        const int8_t *mat = opt->mat;
+        if (opt->meth_mode && seq_[c->seqid].meth_orig_seq != NULL) {
+            if (l_query > meth_qbuf_cap) {
+                meth_qbuf_cap = l_query;
+                meth_qbuf = (uint8_t *) realloc(meth_qbuf, (size_t) meth_qbuf_cap);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)